src/bench/streams_findbyte.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <bench/bench.h>
+
+#include <fs.h>
+#include <streams.h>
+
+static void FindByte(benchmark::Bench& bench)
+{
+    // Setup
+    FILE* file = fsbridge::fopen("streams_tmp", "w+b");
+    const size_t file_size = 200;
+    uint8_t b = 0;
+    for (size_t i = 0; i < file_size; ++i) {
+        fwrite(&b, 1, 1, file);
+    }
+    b = 1;
+    fwrite(&b, 1, 1, file);
```

Comment:
Slightly more concise and readable imo:
```suggestion
    uint8_t data[file_size] = {0};
    data[file_size-1] = 1;
    fwrite(&data, sizeof(uint8_t), file_size, file);
```

Reply:
Less memory efficient, but at a file size of 200 bytes I think optimizing for readability is worthwhile?